flexget/plugins/search/1337x.py

```diff
@@ -0,0 +1,141 @@
+from future.moves.urllib.parse import quote
+from builtins import *  # pylint: disable=unused-import, redefined-builtin
+
+import logging
+import re
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.requests import TimedLimiter, RequestException
+from flexget.utils.requests import Session as RequestSession
+from flexget.utils.soup import get_soup
+from flexget.utils.search import torrent_availability
+from flexget.plugins.plugin_urlrewriting import UrlRewritingError
+
+log = logging.getLogger('1337x')
+
+requests = RequestSession()
+
+class _1337x(object):
+    """
+        1337x search plugin.
+    """
+
+    schema = {
+        'oneOf': [
+            {'type': 'boolean'},
+            {
+                'type': 'object',
+                'properties': {
+                    'order_by': {'type': 'string', 'enum': ['seeders', 'leechers', 'time', 'size'], 'default': 'seeders'}
+                },
+                'additionalProperties': False
+            }
+        ]
+    }
+
+    base_url = 'http://1337x.to/'
+    errors = False
+
+    # urlrewriter API
+    def url_rewritable(self, task, entry):
+        url = entry['url']
+        if url.endswith('.torrent'):
+            return False
+        if url.startswith('http://1337x.to/'):
+            return True
+        return False
+
+    def url_rewrite(self, task, entry):
+        """
+            Gets the download information for 1337x result
+        """
+
+        url = entry['url']
+
+        log.info('1337x rewriting download url: %s' % url)
+
+        try:
+            page = task.requests.get(url)
+            log.debug('requesting: %s', page.url)
+        except RequestException as e:
+            log.error('1337x request failed: %s', e)
+            raise UrlRewritingError('1337x request failed: %s', e)
+
+        soup = get_soup(page.content)
+
+        magnetURL = str(soup.find('a',id='magnetdl').get('href')).lower()
+        torrentURL = str(soup.find('a',id='torrentdl').get('href')).lower()
+
+        entry['url'] = torrentURL
+        entry.setdefault('urls', []).append(torrentURL)
+        entry['urls'].append(magnetURL)
+
+    @plugin.internet(log)
+    def search(self, task, entry, config):
+        """
+            Search for entries on 1337x
+        """
+
+        if not isinstance(config, dict):
+            config = {}
+
+        order_by = ''
+        sort_order = ''
+        if isinstance(config.get('order_by'), str):
+            if config['order_by'] != 'leechers':
+                order_by = '/{0}/desc'.format(config['order_by'])
+                sort_order = 'sort-'
+
+        entries = set()
+
+        for search_string in entry.get('search_strings', [entry['title']]):
+
+            query = '{0}search/{1}{2}/1/'.format(sort_order,quote(search_string.encode('utf8')), order_by)
+            log.debug('Using search params: {0}; ordering by: {1}'.format(search_string, order_by or 'default'))
```

Comment:
I didn't notice this one before. When logging you should let the logger handle the string formatting eg. `log.debug('some logging %s with multiple vars %s', var1, var2)`